### ISSUE TITLE
Write more viewport and scissor tests

### ISSUE THREAD

**Issue by greggman** [KhronosGroup/WebGL#224]
I need to write more scissor and viewport tests.

Specifically,
-   that neither scissor nor viewport change when anything about the current 
  framebuffer changes in including adding/removeing attachments, deleting
  attachments, changing the size of attachments, changing which fbo
  is bound. Some of these tests exist.
-   that scissor can be anything that is not a negative width or negative height.
  In other words
  
  ```
  scissor(-100000, -20000, 300000, 400000)
  ```
  
  is valid
-   That viewport can be set larger than the current backbuffer / fbo and that larger 
  setting correctly effects rendering.
